extensions/hotkey/init.lua

```diff
@@ -176,23 +189,26 @@ function hotkey.new(mods, key, message, pressedfn, releasedfn, repeatfn)
   local keycode = getKeycode(key)
   mods = getMods(mods)
   -- message can be omitted
-  if message==nil or type(message)=='function' then
+  if message==nil or getFunc(message) then
     repeatfn=releasedfn releasedfn=pressedfn pressedfn=message message=nil -- shift down arguments
   end
-  if type(pressedfn)~='function' and type(releasedfn)~='function' and type(repeatfn)~='function' then
+  local f_pressedfn = getFunc(pressedfn)
+  local f_releasedfn = getFunc(releasedfn)
+  local f_repeatfn = getFunc(repeatfn)
+  if not f_pressedfn and not f_releasedfn and not f_repeatfn then
```

Comment:
I think I'd be happier here if 195-197 were:

```lua
pressedfn = getFunc(pressedfn)
releasedfn = getFunc(releasedfn)
repeatfn = getFunc(repeatfn)
```

to reduce the size of the rest of the patch, and eliminate any confusion about which variables are current/correct.